### PR TITLE
[FIX] project(_todo): only internal and active users should be selectable as assignees

### PR DESCRIPTION
**Pull request by bastvdn**
decided to add the domain condition in the project.task user_ids field definition,
which means we can remove identical domains in views
Added empty domains where we could want to add external/inactive user to project.task

Task-3698867

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by MissingNoShiny — APPROVED** [review]
LGTM, thanks!

File: addons/project_todo/views/project_task_views.xml [hunk 1]
```diff
@@ -200,7 +199,7 @@
                                 <field name="name" display="full" class="w-100 text-truncate align-middle"/>
                             </span>
                         </div>
-                        <field name="user_ids" widget="many2many_avatar_user"/>
+                        <field name="user_ids" widget="many2many_avatar_user" domain="[]"/>
```
Comment (xavierbol) [inline addons/project_todo/views/project_task_views.xml:202]: I am not sure to understand why we add a domain attribute here.

Can you explain?

Thanks in advance! 🙂

Comment (bastvdn) [reply addons/project_todo/views/project_task_views.xml:202]: Unlike the other user_ids field, there we actually want to be able to add external/inactive user to the task activity. So we need to specify an empty domain so the default domain doesn't apply

Comment (xavierbol) [reply addons/project_todo/views/project_task_views.xml:202]: Sorry but I don't see why we want that behavior on that view. 😕

Comment (xavierbol) [reply addons/project_todo/views/project_task_views.xml]: Yep probably. 🙂 

thanks for the changes. 🙂

File: addons/project/views/project_task_views.xml [hunk 2]
```diff
@@ -775,7 +774,7 @@
                                     <span t-elif="record.is_private.value" class="fst-italic text-muted"><i class="fa fa-lock"></i> Private</span>
                                 </span>
                             </div>
-                            <field name="user_ids" widget="many2many_avatar_user"/>
+                            <field name="user_ids" widget="many2many_avatar_user" domain="[]"/>
```
Comment (xavierbol) [inline addons/project/views/project_task_views.xml:777]: Same issue here. 😇

**Review by xavierbol — COMMENTED** [review]
Oh it seems we have the same issue here. 🙈

File: addons/project/views/project_task_views.xml (hunk 2 above)
Comment (bastvdn) [reply addons/project/views/project_task_views.xml]: @xavierbol done :grimacing:

**Review by xavierbol — APPROVED** [review]
Youpie! 

LGTM, thanks again for your work! 😁 

robodoo r+

**Pull request closed without merging**